src/base/controller.hpp

```diff
@@ -301,6 +328,34 @@ bool Controller::validate_state(const state_t &state,
   throw std::runtime_error(msg.str());
 }
 
+template <class state_t>
+bool Controller::validate_memory_requirements(state_t &state,
```

Comment:
One of the reasons to calculate real `available memory` in the simulator, is that we can know at runtime whether we have enough memory to run the experiments or not, because we have all the information at this stage. 
Anyway, we have another PR holding (#61) where there's a mechanism to  query available memory in the system, but seems like we want this PR merged before that one, so let's keep this code as is, and refactor it after PR #61 is merged.